### PR TITLE
Fix many typos and links in ADQ component docs and navigation

### PR DESCRIPTION
**Pull request by philsmt**
This fixes countless documentation typos and not properly links in the ADQ digitizer channel component, which I omitted yesterday to get the component merged and ready for a meeting with users today. I also fixed the navigation, which I didn't properly keep up to date in my ignorance.

There's one very small functional change as well in 6c9064b275aa1120f12f287037b065b2c9b7ddad: While checking docs, I encounted one small bug in that the `roi`/`train_roi` passed to `train_data/pulse_data` did not actually work as slices as advertised, but tuple of a slice as if it's passed to `KeyData.ndarray`. Since the shape of digitizer data is known, this seems unneccesarily complex. For ease of implementation, I continued to allow this tuple form and wrap a `slice` into it accordingly.

Finally, there's one setting of `mkdocs` I would like to change in d52a031e5791ef1afbe65fe2db61dcc96b54154b by enabling `show_symbol_type_doc`. I always found it hard in the comparbly long list of some components to distinguish properties from methods. This introduces a visual difference:

image

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -18,19 +18,21 @@ theme:
...
+    - Scans: components/scans.md
+    - Pulse patterns: components/pulse-patterns.md
+    - Delay line detectors: components/delay-line-detectors.md
+    - Optical lasers: components/optical-lasers.md
+    - ADQ Digitizers: components/adq-digitizers.md
+  - karabo-bridge: karabo-bridge.md
+  - Calibration: calibration.md
+  - Signal processing: signal.md
+  - Utilities: utilities.md
+  - Changelog: changelog.md
```
[the remark this answers is not in the document]
Comment (philsmt) [reply mkdocs.yml:35]: I actually changed it on purpose to `karabo-bridge` 🙃 

I required the manual labels for capital `ADQ`, and then decided to keep it for consistency. If you feel strongly to keep the autodetected in identical cases and/or _Karabo bridge_, that's fine by me though too.

[the remark this answers is not in the document]
Comment (JamesWrigley) [reply mkdocs.yml:35]: You can change the name in the navigation section by setting an explicit title:
```md
# ADQ Digitizers

::: extra.components.AdqRawChannel
```

I'd rather do that and keep all the names in one place in their pages.

[the remark this answers is not in the document]
Comment (philsmt) [reply mkdocs.yml]: Ah, TIL!

With that I rolled back the nav changes.

**Review by JamesWrigley — APPROVED** [review]
LGTM!

**Comment by philsmt** [comment]
Thanks for review

**Pull request merged**